.github/workflows/asdf_ci.yml

```diff
@@ -0,0 +1,91 @@
+name: ASDF Continuous Integration
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+env:
+  CRDS_SERVER_URL: "https://jwst-crds.stsci.edu"
+  CRDS_PATH: "/tmp/crds_cache"
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.8
+    - name: Install Dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install flake8
+    - name: Lint with flake8
+      run: |
+        flake8 asdf --count --show-source --statistics
```

Comment:
Basically, all the steps are the same for each job, except for the install command and the tox command.  I'd recommend something like

https://github.com/astropy/astropy/blob/9c79f08d72cdd7982ea990d9b01553ac9c4d657b/.github/workflows/ci_workflows.yml#L67-L88

So that when we have 10 jobs in the matrix, you're not writing 10 sets of steps.  It will be more concise and definitely easier to maintain.